tests/util/tests_BlankNodeIdManagement.cpp

```diff
@@ -0,0 +1,107 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+#include <tuple>
+
+using namespace rdf4cpp::rdf::util;
+using namespace rdf4cpp::rdf;
+
+TEST_SUITE("blank node id management") {
+    TEST_CASE("blank node id generation") {
+        SUBCASE("from entropy") {
+            auto gen = NodeGenerator::new_instance();
+
+            auto const id1 = gen.generate_id();
+            auto const id2 = gen.generate_id();
+            auto const id3 = gen.generate_id();
+
+            CHECK(id1 != id2);
+            CHECK(id2 != id3);
+            CHECK(id1 != id3);
+
+            auto gen2 = NodeGenerator::new_instance();
```

Comment:
After your reply on my comment above I double checked it and found that gen2 is not used. Shoudn't id{4-6} be generated with gen2?

Then the code would actually be the same (except how the generator is instantiated, of course).

Reply:
Almost, the last line is also different